docs/data/date-pickers/date-range-picker/date-range-picker.md

```diff
@@ -81,6 +81,12 @@ This prop will be ignored on the mobile picker.
 
 {{"demo": "DateRangePickerCalendarProp.js"}}
 
+### Conditional props
```

Comment:
I would argue against moving it to the field component page 🤔 I feel like this demo has its place in the customization section here, mostly for discoverability. I don't think it's worth moving it to a different page just because the example targets the field specifically, maybe users will miss the fact that they can do this on the picker. But yeah, it's the same question of where to put demos 🤔 

About the section naming,  I agree that 'Conditional props' is not specific enough, but I would try to keep the title short... maybe something like 'Styling each input differently'?

Reply:
Is there something preventing us from putting it in Custom FIeld ?
I think this page could host everything related to the field customization, and have a link on each component page (because the page title is not 100% transparent).

Reply:
Yeah, just noticed that the `Custom field` is probably the most ideal place for this kind of addition. 👍 🙌 💯

Reply:
And would it make sense for you to add a section in https://mui.com/x/react-date-pickers/date-range-picker/#customization 

### Customize the field

Linking to this page to help people discover it.

Reply:
<img width="857" alt="Screenshot 2023-09-13 at 15 19 55" src="https://github.com/mui/mui-x/assets/4941090/0e84a364-fada-415a-a511-c6295da8b42b">

We already have a section/callout for this.
1. Would you then remove it?
2. Or add the suggested section as an addition?

Reply:
You are right
This is specifically for the single input version

For me we could either:

1. Re-purpose this section to be more general. We rename it, we keep the demo as an example of what's doable in term of customization but we change the text.

2. Create a new section and keep the current one

Reply:
How about dedicating a section to the customization of the range fields in the `Custom fields` page like this: 

- keep the current small section on the [date range picker page](https://mui.com/x/react-date-pickers/date-range-picker/#use-a-single-input-field) illustrating the use of a single input field (IMO, it's necessary info there)
- keep the callout linking to the custom field page, but change the text to something like 'You can also customize the field if you are using a multi input range picker. For more information on field customization, visit....'
- create a section for customizing the range pickers' fields something like 

## Customize the fields on range pickers
### Use single input fields 
-> current section
### Styling multi input fields
-> this demo + whatever we feel like adding in the future

One problem I see with this, however, is that we get back to the issue of jumping from one page to another that people complain about... 😢 But in this case, it makes sense I guess

Reply:
After some experimentation and debates, I've decided to add a new section for it right before the `Validation` section.
WDYT?

Reply:
Make sense to me

Between 
@noraleonte I don't thin "Styling multi input fields" works because some of the customization on this page are also applicable to the single input version (Customize the TextField and Change the format density at least).
Between "Styling" and "Customize", I think "Customize" is broader and covers more the scope of the page.

Reply:
Do we also intend on keeping the `Use a single input field` unchanged [here](https://deploy-preview-10227--material-ui-x.netlify.app/x/react-date-pickers/date-range-picker/#customize-the-field)? I just wonder if there isn't some confusion in linking to the `Custom field` page twice, in two different sections 🤔 
Right now it's 
- Use a single input field - link to custom field page
- Add shortcuts
- Customize the field- link to custom field page
Feels a bit disruptive 🤔

Reply:
How about

- Use a single input field
- Other field customization (link to custom field page)
- Add shortcuts

Reply:
Maybe we could change the wording a bit? 🤔 
WDYT about the following:
`You can find more information in a [dedicated documentation page section](/x/react-date-pickers/custom-field/#use-single-input-fields-on-range-pickers)`

Reply:
That's also fine for me

Reply:
Yeah, changing the wording also sounds good! 👍

Reply:
Updated. 👌 🙏